### PR TITLE
Add repo-test box

### PR DESCRIPTION
**Pull request by alexlarsson**
This stands up repo-test.flathub.org with both a repo-manager and a buildbot master+worker from the branch where its modified to work with the repo-manager.

I also have a `host_vars/repo-test.flathub.org` with the secrets and some other stuff here. Something like:
```
buildbot_master_token: XXXXXXXXXXXXXXXXXXXXXXXXX
github_api_token: XXXXXXXXXXXXXXXXXXXXXX
github_auth_client: XXXXXXXXXXXXXXXXXXXXX
github_auth_secret: XXXXXXXXXXXXXXXXXXXXXXXXXX
github_change_secret: XXXXXXXXXXXXXXXXXXXX
gitlab_change_secret: XXXXXXXXXXXX
buildbot_master_worker_password: XXXXXXXXXXXX
buildbot_worker_master: "127.0.0.1"
buildbot_worker_virtualenv_command: virtualenv-3
buildbot_worker_virtualenv_python: python36
buildbot_worker_password: XXXXXXXXXXXXXXXXXX
buildbot_worker_name: worker1
buildbot_worker_description: Worker on master

```

### CODE REVIEW

**Comment by alexlarsson** [comment]
Note, this has some changes to buildbot-worker which could potentially break the regular debian-based workers, so those need extra careful review.

File: roles/buildbot-master/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  tags:
+   - buildbot-master
+
+- name: Install flathub.gpg
+  get_url:
+    url: https://dl.flathub.org/repo/flathub.gpg
+    checksum: sha256:8bdc20abc4e19c0796460beb5bfe0e7aa4138716999e19c6f2dbdd78cc41aeaa
+    dest: "{{ buildbot_master_dir }}/flathub.gpg"
+    owner: "{{ buildbot_master_user }}"
+    group: "{{ buildbot_master_user }}"
```
Comment (dbnicholson) [inline roles/buildbot-master/tasks/main.yml:88]: Missing `buildbot-master` tag.

**Review by dbnicholson — COMMENTED** [review]
> This is fine for the test, but how would one do this persistent?

(This comment was about persistent UID/GIDs). You add an extra volume to the host and mount it at the relevant location. When deploying a new server you disconnect from the old and connect to the new. Most cloud providers support that usage. Packet does, although it appears mythic doesn't.

The alternative is to restore from a backup after provisioning. That's fairly straightforward for databases since they all have dump utilities. I don't know if buildbot has something similar.

We can punt on this for now, but setting the persistent UID/GIDs now future proofs you in case you do want to manage the persistence at the filesystem level later. I've had to jump through these hoops a few times at Endless where there's persistent data that wasn't considered initially.

**Comment by dbnicholson** [comment]
> I'll switch the two to direct lists, however. I think i need `yum`, not `package` to make the @development-tools group work, no?

We can punt on this, too, but the `package` module is just a wrapper for the appropriate package manager module. So, this would abstract the `yum` vs `dnf` cases, but obviously you wouldn't want to pass `@development-tools` to `apt`. So, this is fine the way it is, but see the package lists in `roles/base/defaults.yml` for an example of building the list depending on the OS. Although, that still splits things by calling `package` from an OS-specific tasks file with the lists concatenated. So, meh, let's punt on this. There's a way to do it, but it's not worth the trouble right now.

**Comment by dbnicholson** [comment]
OK, so I went through the old comments and the new commits and I think I just have one nitpick comment remaining about a missing `buildbot-master` tag. After that I think we can merge.

File: roles/buildbot-master/tasks/main.yml (hunk 1 above)
Comment (alexlarsson) [reply roles/buildbot-master/tasks/main.yml:88]: fixed

**Review by dbnicholson — APPROVED** [review]
Looks good. We can tackle the persistent UIDs when moving this to production (and figure out the secrets more robustly).

**Pull request merged**